src/main/tut/2.4-representable-functors.md

```diff
@@ -0,0 +1,223 @@
+```Haskell
+C(a, x) -> C(a, y)
+```
+```scala
+C[A, X] => C[A, Y]
+```
+................
+```Haskell
+f ∘ h :: a -> y
+```
+```scala
+def `f ∘ h`[A, Y]: A => Y
+```
+................
+```Haskell
+type Reader a x = a -> x
+```
+```tut:silent
+type Reader[A, X] = A => X
+```
+................
+```Haskell
+instance Functor (Reader a) where
+    fmap f h = f . h
+```
+```scala
+implicit def readerFunctor[A] =
+  new Functor[({type T[X] = Reader[A, X]})#T] {
+    def fmap[X, B](f: X => B)(h: Reader[A, X])
+        : Reader[A, B] =
+      f compose h
+  }
+
+/* type lambda with Kind Projector plugin:
+   new Functor[Reader[A, ?]] {...} */
+```
+................
+```Haskell
+type Op a x = x -> a
+```
+```tut:silent
+type Op[A, X] = X => A
+```
+................
+```Haskell
+instance Contravariant (Op a) where
+    contramap f h = h . f
+```
+```scala
+implicit def opContravariant[A] =
+  new Contravariant[({type T[X] = Op[A, X]})#T] {
+    def contramap[X, B](f: B => X)(h: Op[A, X])
+        : Op[A, B] =
+      h compose f
+  }
+
+/* type lambda with Kind Projector plugin:
+   new Contravariant[Op[A, ?]] {...} */
+```
+................
+```Haskell
+instance Profunctor (->) where
+  dimap ab cd bc = cd . bc . ab
+  lmap = flip (.)
+  rmap = (.)
+```
+```scala
+implicit def arrowProfunctor =
+  new Profunctor[Function1] {
+    def dimap[A, B, C, D]
+        (ab: A => B)(cd: C => D)(bc: B => C)
+        : A => D =
+      cd compose bc compose ab
+
+    def lmap[A, B, C]
+        (f: C => A)(ab: A => B): C => B =
+      f andThen ab
+
+    def rmap[A, B, C]
+        (f: B => C)(ab: A => B): A => C =
+      f compose ab
+  }
+```
+................
+```Haskell
+C(-, =) :: Cop × C -> Set
+```
+```scala
+val `C(-, =)`: (Cop `×` C) => Set
+```
+................
+```Haskell
+αx :: C(a, x) -> F x
+```
+```scala
+def α[X]: C[A, X] => F[X]
+```
+................
+```Haskell
+alpha :: forall x. (a -> x) -> F x
+```
+```scala
+/*
+In order to make a universal transformation,
+another type needs to be introduced.
+FunctionK (~>) transforms values from one
+first-order-kinded type into another:
```

Comment:
`first-order-kinded type` is confusing. Also, the description would also apply to
`def foo[F[_], G[_], A](fa: F[A]): G[A]`, which isn't a natural transformation: it's really the higher rank polymorphism that matters here. Personally I'd just remove the second paragraph and link at the docs for FunctionK